data-otservbr-global/scripts/spells/attack/divine_grenade.lua

```diff
@@ -51,36 +53,47 @@ function onTargetCreature(creature, target)
 
 	local position = target:getPosition()
 	addEvent(explodeGrenade, 3000, position, creature:getId())
+	addEvent(removeGrenadeEffect, 3000, position)
 	return true
 end
 
+function removeGrenadeEffect(position)
+	position:removeMagicEffect(CONST_ME_DIVINE_GRENADE)
+end
+
 combatCast:setCallback(CALLBACK_PARAM_TARGETCREATURE, "onTargetCreature")
 
 local spell = Spell("instant")
 
 function spell.onCastSpell(creature, var)
-	if not (creature and creature:isPlayer()) then
+	if not creature and creature:isPlayer() then
```

Comment:
this is not equivalent to what it was before, and I think is incorrect now:

`not creature` and `creature:isPlayer()` would cause a `nil` creature to run `creature:isPlayer()` which would in turn be invalid.
```suggestion
	if not (creature and creature:isPlayer()) then
```

Reply:
updated it to the same logic as Exec Throw